Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -11555,42 +11556,36 @@ void WebPageProxy::setOverlayScrollbarStyle(std::optional<WebCore::ScrollbarOver
         legacyMainFrameProcess().send(Messages::WebPage::SetScrollbarOverlayStyle(scrollbarStyleForMessage), internals().webPageID);
 }
 
-std::optional<Vector<uint8_t>> WebPageProxy::getWebCryptoMasterKey()
+void WebPageProxy::getWebCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
 {
-    if (auto keyData = m_websiteDataStore->client().webCryptoMasterKey())
-        return keyData;
-    if (auto keyData = m_navigationClient->webCryptoMasterKey(*this))
-        return Vector(keyData->span());
-    return std::nullopt;
+    if (m_navigationClient->legacyWebCryptoMasterKey(*this, WTFMove(completionHandler)))
+        return;
+
+    m_websiteDataStore->client().webCryptoMasterKey(WTFMove(completionHandler));
```

Comment:
Use-after-move on `completionHandler`?

Reply:
Had a chat with @achristensen07, and he believes that although syntactically it is a UAM, but the function will early return if the completionHandler is called inside `legacyWebCryptoMasterKey`.

Regardless, I have refactored it out and made it nested so that now the UAM does not exist.